timdex_dataset_api/dataset.py

```diff
@@ -167,12 +167,12 @@ def load(
         if current_records:
             timdex_run_manager = TIMDEXRunManager(timdex_dataset=self)
 
-            # if filters.source is set, further limit to only this source
-            source = filters.get("source")
-            if source:
-                self.paths = timdex_run_manager.get_current_source_parquet_files(source)
-            else:
-                self.paths = timdex_run_manager.get_current_parquet_files()
+            # update paths, limiting by source if set
+            self.paths = timdex_run_manager.get_current_parquet_files(
+                source=filters.get("source")
+            )
+
+            # reload pyarrow dataset
```

Comment:
Even though this block changed in this commit, still curious about your thoughts on my previous comment!

Reply:
See response above!